lib/guest_installation_and_configuration_base.pm

```diff
@@ -401,9 +402,12 @@ sub prepare_non_transactional_environment {
         virt_autotest::utils::setup_rsyslog_host($common_log_folder);
         my $_packages_to_check = 'wget curl screen dnsmasq xmlstarlet yast2-schema python3 nmap';
         zypper_call("install -y $_packages_to_check");
-        my $_patterns_to_check = 'kvm_server kvm_tools';
-        $_patterns_to_check = 'xen_server xen_tools' if ($self->{host_virt_type} eq 'xen');
-        zypper_call("install -y -t pattern $_patterns_to_check");
+        # There is already the highest version for kvm/xen packages on TW
+        if (is_sle) {
```

Comment:
I don't think this is correct as the versions of packages or patterns isn't depending on the distribution type. How about

```suggestion
        if (is_sle('<15-SP5') || is_leap('<15.5')) {
```

Reply:
No, If a fixed version would be used at here, we will have to update it when we are in SLE-next. Meanwhile on the latest SLE/Leap(fg. 15SP5 for now), a 'zypper in/update' usually does not cause troubles.

I remove it from TW because zypper failed sporatically from download.opensuse.org lately.

Reply:
> No, If a fixed version would be used at here, we will have to update it when we are in SLE-next. Meanwhile on the latest SLE/Leap(fg. 15SP5 for now), a 'zypper in/update' usually does not cause troubles.
> 
> I remove it from TW because zypper failed sporatically from download.opensuse.org lately.

So you are not calling zypper on Tumbleweed because the mirroring infrastructure of openSUSE has problems? That does not sound right. Tumbleweed and Leap use the same download.opensuse.org. Then I suggest you check if the necessary version of packages is installed. If they are already there, regardless if it's Tumbleweed or Leap or SLE you can skip the installation.

Reply:
Obviously TW always has the latest packages installed. Meanwhile, It would not helpful to check the versions as access to download.opensuse.org via zypper is still required.
Leap can be skipped as we have not tests for Leap yet.